src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -1314,8 +1337,1089 @@ public boolean mapSwitchControlValue(String value) {
+				populateApplyChangeAndCancelButtonForStream(stats, advancedControllableProperties, streamID);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case MULTICAST_ADDRESS:
+				streamInfo.setAddress(value);
+				this.localStreamInfoList.set(streamIndex, streamInfo);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.MULTICAST_ADDRESS.getName(), streamInfo.getAddress()));
+				populateApplyChangeAndCancelButtonForStream(stats, advancedControllableProperties, streamID);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case SOURCE_ADDRESS:
+				streamInfo.setSourceIp(value);
+				this.localStreamInfoList.set(streamIndex, streamInfo);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.SOURCE_ADDRESS.getName(), streamInfo.getSourceIp()));
+				populateApplyChangeAndCancelButtonForStream(stats, advancedControllableProperties, streamID);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case SOURCE_PORT:
+				streamInfo.setSourcePort(Integer.parseInt(value));
+				this.localStreamInfoList.set(streamIndex, streamInfo);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createNumeric(stats, streamControllingGroup + StreamControllingMetric.SOURCE_PORT.getName(), checkForNullData(streamInfo.getSourcePort().toString())));
+				populateApplyChangeAndCancelButtonForStream(stats, advancedControllableProperties, streamID);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case DESTINATION_PORT:
+				streamInfo.setPort(Integer.parseInt(value));
+				this.localStreamInfoList.set(streamIndex, streamInfo);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createNumeric(stats, streamControllingGroup + StreamControllingMetric.DESTINATION_PORT.getName(), checkForNullData(streamInfo.getPort().toString())));
+				populateApplyChangeAndCancelButtonForStream(stats, advancedControllableProperties, streamID);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case FEC_RTP:
+				FecRTP fecRTP = FecRTP.getByName(value);
+				List<String> fecRTPList = FecRTP.getFecRTPList();
+				streamInfo.setFecRtp(fecRTP.getCode());
+				this.localStreamInfoList.set(streamIndex, streamInfo);
+				addAdvanceControlProperties(advancedControllableProperties, createDropdown(stats, streamControllingGroup + StreamControllingMetric.FEC_RTP.getName(), fecRTPList, fecRTP.getName()));
+				populateApplyChangeAndCancelButtonForStream(stats, advancedControllableProperties, streamID);
+
+				populateLocalExtendedStats(stats, advancedControllableProperties);
```

Comment:
it doesn't need a break statement, does it? It will fall down to the SRT_MODE, is this the expected behaviour?

Reply:
I fixed it.